### PR TITLE
do not loop over all three dimensions, one enough

### PR DESCRIPTION
**Pull request by ferdymercury**

### CODE REVIEW

File: mpl_volume_viewer.py [hunk 1]
```diff
@@ -107,23 +107,22 @@ def process_mouse_button(self, event):
             final_dim = self.index[dim]
             point.insert(dim, final_dim)
             point = [int(round(i)) for i in point]
-            self.set_viewpoint(point)
+            self.set_viewpoint(point, dim)
```
Comment (jni) [inline mpl_volume_viewer.py]: This is not correct when clicking: you actually want two dims when clicking (the ones *not* covered by `dim`.) ie if I click on the xy view I will need to adjust both the xz and the yz views.

I suggest instead that you add `dims=None` to `set_viewpoint`, and then use `dims=[dim]` where you know you only need one dim adjusted (as in `_dscroll`). When `dims=None` (default), it defaults to `range(3)`.

**Review by jni — CHANGES_REQUESTED** [review]
@ferdymercury thanks for all your contributions! 😊 This one isn't yet ready — see my comment below. How does it sound?

File: mpl_volume_viewer.py (hunk 1 above)
Comment (ferdymercury) [reply mpl_volume_viewer.py]: You are right. I have modified it now.

File: mpl_volume_viewer.py [hunk 2]
```diff
@@ -107,12 +107,12 @@ def process_mouse_button(self, event):
             final_dim = self.index[dim]
             point.insert(dim, final_dim)
             point = [int(round(i)) for i in point]
-            self.set_viewpoint(point)
+            self.set_viewpoint(point, [1,2] if dim == 0 else [0,2] if dim == 1 else [0,1] if dim == 2)
```
Comment (jni) [inline mpl_volume_viewer.py]:
```suggestion
            self.set_viewpoint(point, list({0, 1, 2} - {dim}))
```

... for example :smiley:

File: mpl_volume_viewer.py [hunk 3]
```diff
@@ -107,12 +107,12 @@ def process_mouse_button(self, event):
...
 
-    def set_viewpoint(self, point):
+    def set_viewpoint(self, point, dims = None):
```
Comment (jni) [inline mpl_volume_viewer.py]:
```suggestion
    def set_viewpoint(self, point, dims=range(3)):
```

This is needed if you are not going to check for `None` below.

**Pull request merged**